data/transactions/logic/eval.go

```diff
@@ -2977,6 +2977,30 @@ func opGloads(cx *EvalContext) {
 	cx.stack[last] = scratchValue
 }
 
+func opGloadss(cx *EvalContext) {
+	last := len(cx.stack) - 1
+	prev := last - 1
+
+	gi := cx.stack[prev].Uint
+	if gi >= uint64(len(cx.TxnGroup)) {
+		cx.err = fmt.Errorf("gloads lookup TxnGroup[%d] but it only has %d", gi, len(cx.TxnGroup))
+		return
+	}
+	scratchIdx := cx.stack[last].Uint
+	if scratchIdx >= 256 {
+		cx.err = fmt.Errorf("gloadss scratch index >= 256 (%d)", scratchIdx)
+		return
+	}
```

Comment:
It seems `opGloadImpl` does these checks as well, though with slightly different types? There may be a benefit to unifying these checks at some point, since that seems odd to me.

Reply:
^ it looks alike with inputs from different places (and different types I suppose), would be nice to unify the checks too

Reply:
I think that, with some care, we could eliminate some repeat code.  The "rule" right now is that any time a type is being narrowed before sending it to the Impl function, it is tested ahead of time.  But, since unnarrowed types are *not* tested ahead of time, the Impl repeats the test.  We could probably use wider types for the Impl() function, and only do the tests in there.